eth/handler.go

```diff
@@ -809,7 +811,7 @@ type EthNodeInfo struct {
 func (self *ProtocolManager) NodeInfo() *EthNodeInfo {
 	return &EthNodeInfo{
 		Network:    self.networkId,
-		Difficulty: self.blockchain.GetTd(self.blockchain.CurrentBlock().Hash()),
+		Difficulty: self.blockchain.GetTd(self.blockchain.CurrentBlock().Hash(), self.blockchain.CurrentBlock().NumberU64()),
```

Comment:
This is a race condition. Current block might change between two invocations, causing td to be nil. Please query the current block once and fetch the hash/number from that.